tests/common/plugins/loganalyzer/loganalyzer.py

```diff
@@ -170,7 +171,28 @@ def _setup_marker(self, log_files=None):
         self.ansible_host.command(cmd)
         return start_marker
 
-    def analyze(self, marker, fail=True):
+    def disable_log_rotate(self):
+        self.ansible_host.command("sed -i 's/^/#/g' /etc/cron.d/logrotate")
+
+        logging.debug("Waiting for logrotate from previous cron task run to finish")
+        # Wait for logrotate from previous cron task run to finish
+        end = time.time() + 60
+        while time.time() < end:
+            # Verify for exception because self.ansible_host automatically handle command return codes and raise exception for none zero code
+            try:
+                self.ansible_host.command("pgrep -f logrotate")
+            except Exception:
+                break
+            else:
+                time.sleep(5)
+                continue
+        else:
+            logging.error("Logrotate from previous task was not finished during 60 seconds")
+
+    def enable_log_rotate(self):
```

Comment:
If the user will call this function without calling first 'disable' or will call it twice- this will end up with a corrupted /etc/cron.d/logrotate file, as the original comments will be uncommented as well.

Reply:
Good catch. Added a protection.

Reply:
Updated the protection, please refresh before review.